### PR TITLE
feat: mark @BuildStep-annotated methods as implicitly used

### LINKED ISSUE #1095: `@BuildStep`s in an extension processor are considered unused methods
`@BuildStep`s in an extension processor are considered unused methods in IJ, which is not true

_Originally posted by @gastaldi in https://github.com/redhat-developer/intellij-quarkus/issues/1094#issuecomment-1672024387_

### PR DESCRIPTION
**Pull request by fbricon**
fixes #1095 

Before:
Screenshot 2023-08-10 at 15 32 45

After:
Screenshot 2023-08-10 at 16 04 10

@gastaldi  IJ automatically finds usages in .dot file

### CODE REVIEW

**Comment by gastaldi** [comment]
Nice, but the `dot` file is generated only if the system property is set during the build

**Comment by fbricon** [comment]
well yes, but I have no other way at the moment

**Comment by angelozerr** [comment]
@gastaldi is there any chance that your test this PR?

File: src/main/java/com/redhat/devtools/intellij/quarkus/psi/internal/QuarkusBuildImplicitUsageProvider.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import com.redhat.devtools.intellij.lsp4mp4ij.psi.core.jaxrs.JaxRsConstants;
+import com.redhat.devtools.intellij.lsp4mp4ij.psi.core.utils.AnnotationUtils;
+import com.redhat.devtools.intellij.lsp4mp4ij.psi.internal.restclient.MicroProfileRestClientConstants;
+import org.jetbrains.annotations.NotNull;
+
+import static com.redhat.devtools.intellij.lsp4mp4ij.psi.core.utils.AnnotationUtils.hasAnyAnnotation;
+import static com.redhat.devtools.intellij.quarkus.QuarkusConstants.QUARKUS_DEPLOYMENT_BUILDSTEP_ANNOTATION;
+
+/**
+ * Automatically declares as used, methods annotated with {@link io.quarkus.deployment.annotations.BuildStep} annotations,
```
Comment (angelozerr) [inline src/main/java/com/redhat/devtools/intellij/quarkus/psi/internal/QuarkusBuildImplicitUsageProvider.java:28]: Could you give just a sample with BuildStep please.

Comment (gastaldi) [reply src/main/java/com/redhat/devtools/intellij/quarkus/psi/internal/QuarkusBuildImplicitUsageProvider.java:28]: Any Quarkus Extension uses that, for example https://github.com/quarkiverse/quarkus-prettytime/blob/main/deployment/src/main/java/io/quarkiverse/prettytime/deployment/PrettyTimeProcessor.java

Comment (gastaldi) [reply src/main/java/com/redhat/devtools/intellij/quarkus/psi/internal/QuarkusBuildImplicitUsageProvider.java:28]: Sure, just pointing out a sample to be included there 😀

Comment (fbricon) [reply src/main/java/com/redhat/devtools/intellij/quarkus/psi/internal/QuarkusBuildImplicitUsageProvider.java]: done

**Comment by angelozerr** [comment]
> Nice, but the dot file is generated only if the system property is set during the build

Is it easy to retrive this dot file? What about the format of this dot file? Is it big?

**Comment by gastaldi** [comment]
> Is it easy to retrieve this dot file? What about the format of this dot file? Is it big?

You need to build the application with the `-Djboss.builder.graph-output` parameter pointing to a file. It can be large, depending on how many extensions your application uses

**Comment by gastaldi** [comment]
> @gastaldi is there any chance that your test this PR?

Just tested with `./gradlew runIde` and seems to work as advertised. Good job!

**Comment by fbricon** [comment]
Thanks for checking @gastaldi

**Pull request merged**